api/server/runner_httptrigger.go

```diff
@@ -112,15 +112,23 @@ func (trw *triggerResponseWriter) WriteHeader(statusCode int) {
 		realHeaders[k] = vs
 	}
 
-	// XXX(reed): simplify / add tests for these behaviors...
-	gatewayStatus := 200
-	if statusCode >= 400 {
-		gatewayStatus = 502
-	} else if fnStatus > 0 {
-		gatewayStatus = fnStatus
+	trw.inner.WriteHeader(determineFinalStatus(statusCode, fnStatus))
+}
+
+func determineFinalStatus(serviceStatus, userFnStatus int) int {
+	finalStatus := 200
+
+	if userFnStatus > 0 {
+		if userFnStatus >= 500 {
```

Comment:
I guess 400 made sense to me, since we're kind of acting as a gateway in this case on the function's behalf, if the function gives us a 400 back on the actual function response (note: NOT `Fn-Http-Status`), it means fn gave it a botched request of some kind and it would translate to a server side error I think? it's really confusing anyway and I don't think we have a real case for it at the moment anyway, either way.

Reply:
actually, I take this all back. I don't think we should rewrite user status codes at all, it kind of defeats the whole point. I was thinking this was the function status code, not the user status code, which makes sense to futz with. I think this was a mistranslation from previous code, user codes should not be rewritten, and in the previous case the container returning a status code overrode the function status code (this part makes less sense to me, if the user manages to get a status out of there we should probably write it, but it may be the case that they didn't get a chance to write the body as they wanted, so maybe not? ugh)